### DISCUSSION TITLE
SSerialReceived  with Scripts - why doesn't it work

### DISCUSSION THREAD

**Discussion by letshin** [arendst/Tasmota#17548] [Q&A] [answered]
Hi - I've been trying the below but am not able to receive anything by calling SSerialReceived#Data. This previously worked with rules (having set SBaudrate, SerialDelimiter, and SSerialConfig), is there another way of doing things in scripts?
Thanks.

```
>D 20
;define variables here
sen1=0
sen2=12
>T 
;Telemetry
sen1=ANALOG#Range
sen2=SSerialReceived#Data
>S 
;Execute every second
dt [c1l1f1s2p20] %sen2%
>W 
;Write to webUI
Test value1: %sen1%
Test value2: %sen2%
```

**Comment by gemu2015** [comment, marked as answer]
the >T section gets tele period messages, however the serial received data is an event and should require the >E section

**Reply by letshin** [reply]
Thanks - did as you've suggested:
```
>E
;Events
sen2=SSerialReceived#Data
print EVENT
```
but am still not getting any readings, output is this from the console:
```
18:33:43.373 RSL: RESULT = {"SSerialReceived":" z 00015"}
18:33:43.376 EVENT
18:33:43.378 sen2 is 12.00
```

I'm not sure if its worth mentioning that the sensor is hooked up to d3 and d4 of the Wemos D1 and its basically sending data out twice a second which is shown on the console.

**Reply by gemu2015** [reply]
check the JSON message which is read by the script.
there is no # Data in the JSON,  and it is a string.
so read a string.
str=SSerialReceived

**Reply by letshin** [reply]
I tried this for rules and SSerialReceived didn't work but SSerialReceived#Data did, so I did the same here. Andddd, after using just SSerialReceived, this now works. Thanks a lot!
```
18:44:07.098 RSL: RESULT = {"SSerialReceived":" z 00015"}
18:44:07.102 EVENT
18:44:07.103 sen2 is  z 00015
```
